<!DOCTYPE html>
<html style="background:#E2F7CD;">
    <head>
        <meta charset="utf-8">
        <title>3DPatch (help)</title>
        <link href="https://fonts.googleapis.com/css?family=Source+Sans+Pro" rel="stylesheet">
    </head>
    <body style="background:#F2FDE7;width:720px;padding:10px 10px;border-radius:5px;font-family:'Source Sans Pro',sans-serif;box-shadow:0px 0px 5px black;">
        <header>
            <h1>3DPatch (help)</h1>
        </header>
        <main>
            <section>
                <h3>Introduction</h3>
                <p>3DPatch is an application facilitating residue-level information content calculations, homology searches, and 3D structure conservation level-based mark-up for protein sequence or profile hidden Markov model (HMM) queries.</p>
                <p>This section aims to provide an overview of how 3DPatch operates. Following sections provide more detail on individual steps in the data processing workflow and various website components.</p>
                <p>3DPatch can accept either a single protein sequence or a profile HMM file as a new search <a href="#inputHelpSection">input</a>. Alternatively, results of previous 3DPatch sessions can be loaded from a special save point file. Given that the profile HMM file input route simply skips the first few steps of the sequence input-processing pipeline, this section illustrates most of the server functionality on an example utilizing the sequence input route. Loading the results of previous 3DPatch sessions is covered in a <a href="#savePointHelpSection">separate section</a>.</p>
                <p>After entering a valid protein sequence and clicking the submit button, the sequence is processed by the <a href="https://www.ebi.ac.uk/Tools/hmmer/">HMMER web server</a>. There, a <i>phmmer</i> search against the <a href="https://www.uniprot.org/help/reference_proteome">UniProt reference proteomes</a> sequence library is performed. In order to do this, the sequence input is initially converted to a profile HMM featuring position-independent <a href="#searchParametersHelpSection">parameters</a>. HMMER web server performs a search using this profile HMM to find regions of sequences from the UniProt reference proteomes similar to the original sequence input.</p>
                <p>Multiple sequence alignment (MSA) is obtained from the set of alignments of significant UniProt reference proteomes sequence hits to this primitive profile HMM. Second, higher-quality profile HMM is then built from this MSA, with parameters reflecting the frequencies of amino acid residues observed at individual positions in the MSA.</p>
                <p>The profile HMM with position-specific parameters is used as an input to the <i>hmmsearch</i> program, which is used to find regions of sequences corresponding to 3D structures from the <a href="https://www.ebi.ac.uk/pdbe/">Protein Data Bank (PDB)</a> similar to the original sequence input. At the same time, information content at individual positions in the profile HMM is calculated using the <a href="https://skylign.org/">Skylign</a> server. If a profile HMM file is used as an input to 3DPatch, previous steps are skipped and the submitted profile HMM is used directly as an input to <i>hmmsearch</i> and Skylign.</p>
                <p>Mapping between individual residues in the identified PDB structures and positions in the profile HMM is obtained from the alignment of sequences corresponding to the respective 3D structures to the profile HMM. These alignments can be visualized in the <a href="#sequenceAlignmentHelpSection">sequence alignment</a> window. <a href="#hmmCoverageHelpSection">Coverage of profile HMM positions by the PDB structures</a> is illustrated using a few informative structures, and a list of other PDB structures containing a hit to the profile HMM is generated. Any of these structures can then be visualized with residue-level information content-based mark-up using the <a href="https://webchemdev.ncbr.muni.cz/Litemol/">LiteMol</a> viewer.</p>
                <p>The input-processing workflow described above can be summarized in the following scheme:<img width="720" height="405" alt="scheme" src="help-images/scheme.png"></p>
            </section>
            <section id="inputHelpSection">
                <h3>Input</h3>
                <p>Input to 3DPatch can be either a single protein sequence or a profile HMM file. These inputs must be in formats recognized by the immediately called HMMER programs (<i>phmmer</i> or <i>hmmsearch</i>, respectively). Refer to the <a href="https://hmmer-web-docs.readthedocs.io/en/latest/searches.html">HMMER web server documentation</a> for detailed information on format requirements.</p>
                <p>When sequence input is used, all whitespace characters are removed from the sequence before <i>phmmer</i> is called. If a header line beginning with the <span style="font-family:monospace">&gt;</span> symbol is found preceding the sequence, then the line is ignored.</p>
            </section>
            <section id="searchParametersHelpSection">
                <h3>Search parameters</h3>
                <p>All search parameters currently take on default values selected by the HMMER programs, the HMMER web server, and the Skylign server. HMMER web server default settings take precedence over those used by HMMER programs.</p>
                <p>In the case of the sequence input route, the initial <i>phmmer</i> search against the UniProt reference proteomes uses a profile HMM with position-independent parameters derived from the BLOSUM62 substitution scoring matrix. In this profile HMM, the position-independent gap opening and gap extension probabilities are set to 0.02 and 0.4, respectively.</p>
                <p>Per-target and per-domain reporting E-value thresholds are set to 1.0 for both <i>phmmer</i> and <i>hmmsearch</i>. Per-target and per-domain inclusion (significant hit) E-value thresholds are set to 0.01 and 0.03, respectively.</p>
                <p>No taxonomy-based filtering of hits is being performed at any stage.</p>
            </section>
            <section id="savePointHelpSection">
                <h3>Saving and loading state</h3>
                <p>Save point files can be created once necessary information is available. All save point files contain the profile HMM information content profile, the list of PDB structures illustrating coverage of the profile HMM, and several variables necessary for state recreation. Optionally, if a structure was visualized at the time of save point file creation, it will be immediately shown in the LiteMol plugin after loading. Note that it is also possible to save the view of a structure not included in the profile HMM structure coverage illustration this way.</p>
                <p>Loading state from a save point file recreates the profile HMM information content profile graph, the profile HMM structure coverage illustration, and the sequence alignment window. Optionally, the previously selected 3D structure is visualized.</p>
                <p>It is possible to select a new structure from the profile HMM structure coverage illustration for information content-based mark-up when loading state from a save point file.</p>
                <p>Some actions are disabled when loading state from a save point file. New search using a sequence or a profile HMM file input must be performed to have them enabled. These actions are:<ul>
                    <li>Selection of 3D structures not included in the profile HMM structure coverage illustration, or, optionally, the structure selected from the drop-down list of other domain hits at the time of save point file creation.</li>
                    <li>Changing the color scaling option.</li>
                    <li>Creation of new save point files.</li>
                </ul></p>
            </section>
            <section id="sequenceAlignmentHelpSection">
                <h3>Sequence alignment</h3>
                <p>Contents of this window vary depending on the type of input. If sequence input was used, then the consensus sequence representing the profile HMM built from the <i>phmmer</i> search results, the input sequence aligned to this profile HMM, and optionally the alignment of the selected PDB chain sequence region will be shown. If a profile HMM file input was used, then there will be no input sequence alignment.</p>
                <p>The profile HMM consensus sequence is shown on the line starting with <span style="font-family:monospace">CONS</span>. It contains one residue symbol per profile HMM position. The consensus sequence is read either from the profile HMM built by the HMMER web server from the <i>phmmer</i> search results, or from the uploaded profile HMM file.</p>
                <p>If sequence input route was utilized, then the alignment of the input sequence to the profile HMM built from the <i>phmmer</i> search results is shown on the line starting with <span style="font-family:monospace">INPUT</span>.</p>
                <p>If some PDB structure is selected from the profile HMM structure coverage window for information content-based mark-up, the sequence of the marked-up region aligned to the profile HMM, and optionally to the input sequence, is shown on the line starting with the PDB chain ID. User can hover the mouse pointer over individual residues in this sequence to see them highlighted in the LiteMol viewer 3D structure visualization. When the selected residue is present in the 3D structure, its background color changes to magenta; if it is missing from the 3D structure, the background color changes to blue.</p>
                <p>Note that residues assigned to insert states are not shown in the sequence alignment window.</p>
            </section>
            <section id="hmmCoverageHelpSection">
                <h3>Profile HMM structure coverage</h3>
                <p>When <i>hmmsearch</i> is called and the PDB is used as a sequence database, a list of PDB chains containing a hit to the profile HMM is returned. Each PDB chain can contain multiple sequence regions (called <i>domains</i> in HMMER terminology and in this section) aligned to the profile HMM. A score is calculated for each of these domain alignments. E-value reflecting the scores of individual domain alignments is assigned to each PDB chain. The list of PDB chains returned by <i>hmmsearch</i> is sorted by ascending chain E-value.</p>
                <p>3DPatch goes through the list of PDB chain hits, starting with the chain with the lowest assigned E-value. For each chain, domains recognized within are iterated through in the order of their proximity to the protein N-terminus, starting with the most N-terminal one.</p>
                <p>For each domain, a list profile HMM positions involved in the alignment of the domain sequence to the profile HMM is generated. If the profile HMM structure coverage illustration currently contains no domains, or if the number of new (previously uncovered) profile HMM positions covered by the alignment of the current domain is at least <i>N</i>, then the domain is added to the profile HMM structure coverage illustration, and the list of profile HMM positions covered by 3D structures is updated. The value of <i>N</i> is currently set to 40 positions.</p>
                <p>If a domain does not meet the condition stated above, it does not appear in the profile HMM structure coverage illustration. However, it can still be visualized by manually selecting it from the drop-down list of all other domain hits. Each domain in the list is characterized by the PDB chain ID, the span of chain sequence residues involved in the alignment, and the span of profile HMM positions covered by the alignment. In addition, significance of the domain hit is graphically illustrated based on its independent E-value: &#x1f525; for E-values below 1e-20, *** for E-values between 1e-20 (inclusive) and 1e-10, ** for E-values between 1e-10 (inclusive) and 1e-2, and * for E-values greater than or equal to 1e-2.</p>
                <p><strong>Note</strong> that PDB chains are not treated equally by the HMMER web server. When multiple chains have the same amino acid sequence, they would yield the same domain hits spanning the same sets of residues. Each target considered by <i>hmmsearch</i> corresponds to a unique amino acid sequence, which can be shared by one or more PDB chains. The HMMER web server uses PDB chain ID of one of these chains as the <i>accession</i> attribute of the target sequence. 3DPatch makes it possible to also visualize domains found in chains other than the target's <i>accession</i>. These domains are added to the drop-down menu immediately after the corresponding domain recognized within the <i>accession</i> PDB chain had been added to the profile HMM structure coverage illustration or to the drop-down list. A table of <i>accession</i> PDB chains along with a comma-separated list of other PDB chains sharing the same amino acid sequence can be accessed from the 3DPatch website after the searches had finished. It may not be possible to display this table in some browsers when an advertisement blocking software is installed. It is not possible to select 3D structures from the drop-down menu when state is loaded from a save point file, with the exception of the structure selected at the time of save point file creation.</p>
            </section>
            <section id="colorScalingHelpSection">
                <h3>Color scaling options</h3>
                <p>3DPatch currently implements four information content color scaling options. Each of these options partitions the information content range into ten bins and assigns a color from the color spectrum to all information content values falling to a specific bin. The options differ by the transformation applied to individual information content values and by the value taken as the information content maximum. These options are:<ol>
                    <li><i>Linear (absolute)</i>, which uses maximum theoretical information content (appr. 6.45) as the maximum. Information content range is split evenly into ten bins, each covering around 0.65 bits.</li>
                    <li><i>Linear (relative)</i>, which uses maximum observed information content as the maximum. Information content range is split evenly into ten bins, each covering one tenth of the maximum observed information content value. <strong>Note</strong> that the maximum observed value is calculated across all positions in the profile HMM, not only those covered by the currently visualized PDB structure. This is the default option.</li>
                    <li><i>Exponential (absolute)</i>, which uses <span style="font-family:monospace">exp</span>(maximum theoretical information content), appr. 634.68, as the maximum. The range is split evenly into ten bins, each covering around 63.47 units. Values of information content for which <span style="font-family:monospace">exp</span>(information content) falls into the the range of the bin are assigned the corresponding color. For example, the first color is assigned to positions with information content between <span style="font-family:monospace">log</span>(0) and <span style="font-family:monospace">log</span>(63.47), <i>i.e.</i>, between 0 (<span style="font-family:monospace">-inf</span>) and 4.15 bits.</li>
                    <li><i>Exponential (relative)</i>, which is the same as <i>Exponential (absolute)</i>, but uses <span style="font-family:monospace">exp</span>(maximum observed information content) as the maximum.</li>
                </ol></p>
                <p>The color scaling option used can be changed after the calculations had finished without need to repeat preceding HMMER searches. Note that changing the color scaling option is not allowed when loading state from a save point file.</p>
                <p>Coloring of residues outside the alignment region and residues assigned to insert states (green) and residues belonging to other chains (light blue) is unaffected by the color scaling choice.</p>
            </section>
            <section>
                <h3>Misc</h3>
                <p>It can happen that the 3DPatch process hangs when the HMMER servers are overloaded. You may want to reload the page and repeat the search, or try again later.</p>
                <p>The PDB sequence database used by the HMMER web server may not be fully synchronized with the <a href="https://www.ebi.ac.uk/pdbe/">PDBe</a>. Newly added structures may not appear in the list of hits, and structures made obsolete recently may fail to visualize.</p>
            </section>
        </main>
    </body>
</html>
